crates/credentials/src/verifiable_credential.rs

```diff
@@ -206,64 +252,87 @@ impl VerifiableCredential {
             .ok_or(CredentialError::MissingNbf)?;
         let exp = registered_claims.expiration;
 
-        // TODO: Change this to None after this issue is resolved - https://github.com/TBD54566975/web5-rs/issues/202
-        // check claims match expected values
-        if vc.id == String::default() {
-            vc.id = jti;
-        } else if vc.id != jti {
-            return Err(CredentialError::IdMismatch);
+        if let Some(id) = &vc_payload.id {
+            if id != &jti {
+                return Err(CredentialError::IdMismatch);
+            }
+        }
+
+        if let Some(issuer) = &vc_payload.issuer {
+            let vc_issuer = issuer.to_string();
+            if iss != vc_issuer {
+                return Err(CredentialError::IssuerMismatch);
+            }
         }
 
-        let vc_issuer = vc.issuer.to_string();
-        if vc_issuer.is_empty() {
-            vc.issuer = Issuer::String(iss.clone());
-        } else if iss != vc_issuer {
-            return Err(CredentialError::IssuerMismatch);
+        if let Some(credential_subject) = &vc_payload.credential_subject {
+            if sub != credential_subject.id {
+                return Err(CredentialError::SubjectMismatch);
+            }
         }
 
-        if vc.credential_subject.id == String::default() {
-            vc.credential_subject.id = sub;
-        } else if vc.credential_subject.id != sub {
-            return Err(CredentialError::SubjectMismatch);
+        if let Some(issuance_date) = &vc_payload.issuance_date {
+            if issuance_date != &nbf {
+                return Err(CredentialError::IssuanceDateMismatch);
+            }
         }
 
-        if vc.issuance_date == i64::default() {
-            vc.issuance_date = nbf;
-        } else if vc.issuance_date != nbf {
-            return Err(CredentialError::IssuanceDateMismatch);
+        if let Some(expiration_date) = &vc_payload.expiration_date {
+            if exp.is_none() {
+                return Err(CredentialError::MisconfiguredExpirationDate(
+                    "VC has expiration date but no exp in registered claims".to_string(),
+                ));
+            }
+
+            if expiration_date != &exp.unwrap() {
```

Comment:
ditto https://github.com/TBD54566975/web5-rs/pull/217/files#r1609142601

Reply:
ah i see you're checking `if exp.is_none()`. i'd prefer if we use a `match` statement, because seeing an `unwrap()` outside of tests sets off mental alarm bells.

Reply:
updated to get rid of all unwraps 👍